vendor/sigs.k8s.io/kustomize/api/internal/plugins/builtinconfig/loaddefaultconfig.go

```diff
@@ -33,7 +33,7 @@ func loadDefaultConfig(
 // makeTransformerConfigFromBytes returns a TransformerConfig object from bytes
 func makeTransformerConfigFromBytes(data []byte) (*TransformerConfig, error) {
 	var t TransformerConfig
-	err := yaml.Unmarshal(data, &t)
+	err := yaml.UnmarshalStrict(data, &t)
```

Comment:
which user-facing files will this become more strict on? does this need a release note?

Reply:
This change is strict for [transformerconfig](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/transformerconfigs/README.md).
The `transformerconfig` allows the user to add eligible fields for any transformers in kustomize. The main use case is for the user to edit fields defined by any crds in the kustomize functions.
The reason for this change is that we want to detect user misconfiguration.

Reply:
> does this need a release note?
I don't think so.
Because I think the user configuration was broken when that strict unmarshal failed.